http3/roundtrip.go

```diff
@@ -175,9 +179,13 @@ func (r *RoundTripper) getClient(hostname string, onlyCached bool) (roundTripClo
 		if err != nil {
 			return nil, err
 		}
-		r.clients[hostname] = client
+		r.clients[hostname] = cl
+	}
+	if cl.IsClosed() {
```

Comment:
Don't we have a race condition here, when the client is closed in between here and where it's returned?

Reply:
Yeah. Any suggestion how to fix this? There’s actually a quite long window here between the client being returned and the request being received by the server, in which the connection could die. Seems like the only way to properly fix this is by having some (bounded) retry logic?

Reply:
`when the client is closed in between here and where it's returned` might actually not be a problem, it should get recreated the next time `.getClient` is called. For bounded retry, you can add a parameter `retries` since it is calling itself recursively.